### PR TITLE
theme

### PR DESCRIPTION
**Pull request by kachakot**

### CODE REVIEW

File: src/components/ButtonLink/styles.ts [hunk 1]
```diff
@@ -19,68 +17,76 @@ interface IProps {
...
   display: inline-flex;
   align-items: center;
   flex-shrink: 0;
   width: auto;
-  min-width: ${({ theme }) => theme.btn.width};
-  font-size: ${({ theme }) => theme.fontSize.fontSize15};
-  font-weight: ${({ theme }) => theme.fontWeight.semiBold};
+  min-width: ${variables.width};
+  font-size: 1rem;
+  font-weight: ${({ theme }) => theme.fontWeights.semiBold};
```
Comment (jdakowicz) [inline src/components/ButtonLink/styles.ts]: Sometimes you are using destructurization
`font-weight: ${({ theme }) => theme.fontWeights.semiBold};`
Sometimes you don't
`transition: ${props => props.theme.transitionDuration}ms ease;`
Try to use one convention in the future requests

File: src/components/ButtonLink/styles.ts [hunk 2]
```diff
@@ -19,68 +17,76 @@ interface IProps {
   left?: boolean;
 }
 
+const variables = {
+  width: '10.5rem',
+  heightMedium: '3.375rem',
+  heightBig: '3.75rem',
+  primaryBgHover: '#e72934',
```
Comment (jdakowicz) [inline src/components/ButtonLink/styles.ts]: This values are not stored in a `theme` ?

File: src/components/Container/styles.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import styled from 'styled-components';
+
+import { IColor } from 'themes/variables';
+
+interface IContainerStyledProps {
+  background?: IColor;
+  mt?: string;
```
Comment (jdakowicz) [inline src/components/Container/styles.ts]: If you are using the `styled-system` it should already have these types declared so you can just extend this interface by t
hem something like this but I am not sure what are the names of this interfaces
```
interface IContainerStyledProps extends PaddingProps, MarginProps {
  background?: IColor;
}
```

File: src/components/Container/styles.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  pl?: string;
+  pr?: string;
+  px?: string;
+  py?: string;
+  p?: string;
+  m?: string;
+}
+
+const ContainerStyled = styled.div<IContainerStyledProps>`
+  ${({ m }) => (m ? `margin: ${m}` : '')};
```
Comment (jdakowicz) [inline src/components/Container/styles.ts]: Can you use `styled-system` somehow here ?

File: src/components/StatNumber/styles.ts [hunk 5]
```diff
@@ -1,40 +1,40 @@
...
   position: relative;
   width: 100%;
-  padding-top: ${props => props.theme.gutter.gutter24};
-  padding-bottom: ${props => props.theme.gutter.gutter8};
+  padding-top: 1.5rem;
+  padding-bottom: 0.5rem;
 
   &::before {
     position: absolute;
     z-index: -1;
```
Comment (jdakowicz) [inline src/components/StatNumber/styles.ts]: Indexes should be in a theme too so you can easily manage the layers order in one place

File: src/themes/variables.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
+export type IColor = 'black' | 'white' | 'darkGrey' | 'mediumGrey' | 'lightGrey' | 'red' | 'darkNavy' | 'navy' | 'blue';
+
+export type IFontWeight = 'regular' | 'medium' | 'semiBold' | 'bold';
+
+export type ILineHeight = 'big' | 'medium' | 'small' | 'normal';
+
+export type ITypo = 'typo1' | 'typo2' | 'typo3' | 'typo4' | 'typo5' | 'typo6' | 'typo7';
+
+export interface IThemeInterface {
+  colors: { [color in IColor]: string };
```
Comment (jdakowicz) [inline src/themes/variables.ts]: Nice1 :)

File: src/themes/variables.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  breakpoints: string[];
+  textStyles: { [typo in ITypo]: {} };
```
Comment (jdakowicz) [inline src/themes/variables.ts]: You don't know the structure for the typo yet?

Comment (kachakot) [reply src/themes/variables.ts]: I am not sure yet, I will standardize it on another branch.

**Pull request merged**